File: Pascal'sTriangle.c
/**
 * Return an array of arrays of size *returnSize.
 * The sizes of the arrays are returned as *returnColumnSizes array.
 * Note: Both returned array and *columnSizes array must be malloced, assume caller calls free().
 */

int** generate(int numRows, int* returnSize, int** returnColumnSizes) {
    if (numRows == 0) {
        *returnSize = 0;
        return NULL;
    }

    int** triangle = (int**)malloc(numRows * sizeof(int*));
    *returnColumnSizes = (int*)malloc(numRows * sizeof(int));

    for (int i = 0; i < numRows; i++) {
        triangle[i] = (int*)malloc((i + 1) * sizeof(int));
        (*returnColumnSizes)[i] = i + 1;
        
        for (int j = 0; j <= i; j++) {
            if (j == 0 || j == i) {
                triangle[i][j] = 1;
            } else {
                triangle[i][j] = triangle[i - 1][j - 1] + triangle[i - 1][j];
            }
        }
    }
    
    *returnSize = numRows;
    return triangle;
}

File: Pascal'sTriangle.py
class Solution:
    def generate(self, numRows: int) -> List[List[int]]:
        if numRows == 0:
            return []
        l = [[1]]
        if numRows == 1:
            return l
        else:
            i = 1
            while i < numRows:
                row = [1] * (i+1)
                for j in range(0,i+1):
                    if j == 0:
                        row[j] = 1
                    elif j == i:
                        row[j] = 1
                    else:
                        row[j] = l[i-1][j-1] + l[i-1][j]
                l.append(row)
                i+=1
        return l
